File: backend/src/app.module.ts
import { Module } from '@nestjs/common';
import { AuthModule } from './auth/auth.module';
import { CommonModule } from './common/common.module';
import { CategoriesModule } from './categories/categories.module';
import { ThingsModule } from './things/things.module';

@Module({
  imports: [CommonModule, AuthModule, CategoriesModule, ThingsModule],
})
export class AppModule {}

File: mobile/lib/src/app_module.dart
import 'package:firebase_auth/firebase_auth.dart';
import 'package:flutter_modular/flutter_modular.dart';
import 'package:home_stuff/src/features/auth/auth_module.dart';
import 'package:home_stuff/src/features/splash/ui/pages/splash_page.dart';
import 'package:home_stuff/src/features/things/things_module.dart';

class AppModule extends Module {
  @override
  void binds(Injector i) {
    i.addInstance<FirebaseAuth>(FirebaseAuth.instance);
  }

  @override
  List<Module> get imports => [
        AuthModule(),
      ];

  @override
  void routes(RouteManager r) {
    r.child('/splash', child: (_) => const SplashPage());
    r.module('/things', module: ThingsModule());
    r.module('/auth', module: AuthModule());
  }
}
